### PR TITLE
Fix Issue where we can no longer create zip files on windows during cli tests.

### PR DESCRIPTION
**Pull request by thboop**
The new virtual environment shipped for windows today which resulted in a change where our powershell script could no longer create the directory on windows. We should just move to pwsh (powershell core) as there is more support for that shell.

### CODE REVIEW

File: packages/tool-cache/__tests__/tool-cache.test.ts [hunk 1]
```diff
@@ -553,7 +553,7 @@ describe('@actions/tool-cache', function() {
       if (IS_WINDOWS) {
         const escapedStagingPath = stagingDir.replace(/'/g, "''") // double-up single quotes
         const escapedZipFile = zipFile.replace(/'/g, "''")
-        const powershellPath = await io.which('powershell', true)
```
Comment (ericsciple) [inline packages/tool-cache/__tests__/tool-cache.test.ts:556]: consider `await io.which('pwsh', false) || await io.which('powershell', true)`

**Comment by mishu017** [comment]
[]()

File: packages/tool-cache/__tests__/tool-cache.test.ts (hunk 1 above)
Comment (mishu017) [reply packages/tool-cache/__tests__/tool-cache.test.ts:556]: #- [__]()

Comment (mishu017) [reply packages/tool-cache/__tests__/tool-cache.test.ts:556]: No []()

**Pull request merged**